### PR TITLE
rustbuild: distribute cargo-fmt alongside rustfmt

### PR DESCRIPTION
**Pull request by Keruspe**
Not sure whether we want that nor if it's the right way to do so, but it feels quite weird to have rustfmt without cargo-fmt. Or are there other plans wrt that?

What do you think @nrc ?

### CODE REVIEW

**Comment by nrc** [comment]
Thanks! Seems good to me, but r? @Mark-Simulacrum for the build system changes.

File: src/bootstrap/dist.rs [hunk 1]
```diff
@@ -1163,7 +1163,12 @@ impl Step for Rustfmt {
             compiler: builder.compiler(stage, build.build),
             target
         }).expect("Rustfmt to build: toolstate is testing");
+        let cargofmt = builder.ensure(tool::Cargofmt {
+            compiler: builder.compiler(stage, build.build),
+            target
+        }).expect("Rustfmt to build: toolstate is testing");
```
Comment (Mark-Simulacrum) [inline src/bootstrap/dist.rs]: Please change the expect statement here to "cargofmt" or something like that.

**Review by Mark-Simulacrum — CHANGES_REQUESTED** [review]
Looks pretty good. I'm not too happy with the macro, but I think it's okay for now at least. We do similar things elsewhere, and until we come up with a better solution this should work.

Please fix the one comment and r=me afterwards.

**Comment by Keruspe** [comment]
r= @Mark-Simulacrum

**Comment by Keruspe** [comment]
(I kind of expected that indeed, would have been surprised otherwise)

**Comment by Keruspe** [comment]
@kennytm not waiting on bors yet, fwiw (wrt tags)

**Comment by oli-obk** [comment]
`cargo fmt` could be implemented as a stable `cargo install`able tool, which would not require distributing it, since it would never break.

**Pull request merged**